data-model/src/main/java/io/micronaut/data/model/PersistentPropertyPath.java

```diff
@@ -279,4 +289,26 @@ public Optional<NamingStrategy> findNamingStrategy() {
         }
         return owner.findNamingStrategy();
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        PersistentPropertyPath that = (PersistentPropertyPath) o;
+        return Objects.equals(associations, that.associations) && Objects.equals(property, that.property);
+    }
+
+    @Override
+    public int hashCode() {
+        return property.hashCode();
```

Comment:
Should `hashCode` use `associations` since `equals` uses it?

Reply:
Doesn't need to be. In this case I don't expect multiple paths with the same property.